src/Context/FixtureContext.php

```diff
@@ -493,7 +493,14 @@ public function stepUpdateRecordState($type, $id, $state)
                 Versioned::set_reading_mode($oldMode);
                 break;
             case 'deleted':
-                $obj->delete();
+                if ($obj->hasExtension(Versioned::class)) {
+                    $obj->suppressDeletedVersion(fn () => $obj->delete());
+                } else {
+                    $obj->delete();
+                }
```

Comment:
Needed to fix https://github.com/creative-commoners/silverstripe-installer/actions/runs/15199868196/job/42751924826 using option 1 from https://github.com/silverstripe/silverstripe-versioned/pull/511#issuecomment-2902905039 - any discussion about whether that is the correct option should be held in that PR, not this one.

Note that we explicitly check for the `Versioned` extension here because `the "Some\Class" "record name" is deleted` can be useful for deleting non-versioned records as well - as opposed to `is archived` or `is published` which can only be done on versioned records.

Reply:
No longer needed